packages/guides-restructured-text/src/RestructuredText/Parser/Productions/TitleRule.php

```diff
@@ -142,7 +142,8 @@ public function isSpecialLine(string $line): ?string
     private function currentLineIsAnOverline(string $line, ?string $nextLine): string
     {
         $letter = $this->isSpecialLine($line);
-        if ($nextLine !== null && $letter && $this->isTextLine($nextLine)) {
+        if ($nextLine !== null && $letter && $this->isTextLine($nextLine) &&
+            (strlen($line) > 2 || strlen($line) >= strlen($nextLine))) {
```

Comment:
Please use `mb_strlen` as the title might contain multibyte chars. 

Should the `strlen($line) > 2` not be part of the `isSpecialLine` method? this will make this statement more readable.